src/app/(admin)/admin/_lib/validationSchemas/organizationSchema.js

```diff
@@ -0,0 +1,75 @@
+import { z } from 'zod';
+import {
+  MESSAGES,
+  createValidationSchema,
+  specialistCore,
+  zCreateAddressSchema,
+  zEditAddressSchema,
+  zInteger,
+  zString,
+  zStringArray,
+  zStringWithMax,
+} from './specialistCommonSchemas';
+
+// ------------------ COMMON SECTION ---------------------
+
+const zOrganizationSchema = specialistCore.extend({
+  name: zStringWithMax,
+  yearsOnMarket: zInteger,
+  description: zString,
+});
+
+const restProps = zOrganizationSchema.extend({
+  addresses: zCreateAddressSchema.array().default([]),
+});
+
+// ------------------ CREATE SECTION ---------------------
+
+const createDefaultProps = z.object({
+  name: zStringWithMax,
+  type: zStringArray,
+});
+
+const activeOrganizationSchema = restProps.extend({
+  isActive: z.literal(true),
+});
+
+const draftOrganizationSchema = restProps.partial().extend({
+  isActive: z.literal(false),
+});
+
+export const organizationSchemaUnion = z.discriminatedUnion('isActive', [
+  activeOrganizationSchema,
+  draftOrganizationSchema,
+]);
+export const organizationCreateValidationSchema = createValidationSchema(organizationSchemaUnion, createDefaultProps);
+
+// ------------------ EDIT SECTION ---------------------
```

Comment:
why split edit and create schemas? validation rules are the same for both as I remember

Reply:
Thanks for comment! I our case creating and editing of specialist and organization requires a lil bit different fields to be filled because of form implementations. In one hand, editing of specialist requires `therapiesIds`  to have a least one item in the end, in another hand, creating doesn't require this field at all.